src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,201 @@
+        bobPasswordNull.setPassword(null);
+        bobPasswordNull.setId(52698L);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(bobPasswordNull);
+        }, "An exception should be thrown");
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
+        User bob = new User();
+        bob.setAge(10);
+        bob.setLogin("bobist");
+        bob.setPassword("hhjfbchbv");
+        bob.setId(52698L);
+
+        User alice = new User();
+        alice.setAge(15);
+        alice.setLogin("aliceFox");
+        alice.setPassword("hhjf562bv");
+        alice.setId(36898L);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(bob);
+        }, "An exception should be thrown");
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(alice);
+        }, "An exception should be thrown");
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        User bob = new User();
+        bob.setAge(-10);
+        bob.setLogin("bobist");
+        bob.setPassword("hhjfbchbv");
+        bob.setId(52698L);
+
+        User alice = new User();
+        alice.setAge(-3);
+        alice.setLogin("aliceFox");
+        alice.setPassword("hhjf562bv");
+        alice.setId(36898L);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(bob);
+        }, "An exception should be thrown");
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(alice);
+        }, "An exception should be thrown");
+    }
+
+    @Test
+    void register_invalidLogin_notOk() {
+        User john = new User();
+        john.setAge(19);
+        john.setLogin("bobist");
+        john.setPassword("hhfjbcdyb");
+        john.setId(1259L);
+        registrationService.register(john);
+
+        User bob = new User();
+        bob.setAge(19);
+        bob.setLogin("bobist");
+        bob.setPassword("hiokfbchbv");
+        bob.setId(52698L);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(bob);
+        }, "An exception should be thrown");
+    }
+
+    @Test
+    void register_lengthPassword_notOk() {
+        User bob = new User();
+        bob.setAge(19);
+        bob.setLogin("bobistik");
+        bob.setPassword("hhjf");
+        bob.setId(52698L);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(bob);
+        }, "An exception should be thrown");
+
+        User john = new User();
+        john.setAge(19);
+        john.setLogin("johnidd");
+        john.setPassword("jo");
+        john.setId(1259L);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(john);
+        }, "An exception should be thrown");
+    }
+
+    @Test
+    void register_lengthLogin_notOk() {

```

Comment:
```suggestion
    void register_shortLogin_notOk() {
```
друга частина для опису того, що даємо на вхід тестовому методу